risc0/zkp/rust/src/lib.rs

```diff
@@ -25,6 +25,8 @@ mod merkle;
 #[cfg(feature = "prove")]
 pub mod prove;
 pub mod taps;
+#[cfg(feature = "prove")]
+mod test_circuit;
```

Comment:
One idea is to have a separate `risc0-circuit-fib` crate that is a `dev-dependency` of this crate. That way, in production, users don't need to have the fib circuit making libraries larger than necessary.

Reply:
Hmm, I can't figure out how to make this work from a dependency graph perspective. Specifically, the `test_circuit` module from this PR depends on the `adapter` and `taps` modules of the `risc0-zkp` crate, and the `prover` module of the `risc0-zkp` crate depends on this `test_circuit` module (for tests). This works fine with `test_circuit` as a module of `risc0-zkp`, but if I were to pull it out into a separate `risc0-circuit-fib` crate, it seems like it would make `risc0-zkp` have a dev dependency of `risc0-circuit-fib` which would in turn have a dependency on `risc0-zkp` -- i.e. a circular dependency.

Reply:
`dev-dependencies` are treated differently. You can think of it like:

* risc0-zkp (dev-dep)
* risc0-circuit-fib
* risc0-zkp

Reply:
IOW, a dev-dep is a different node in the graph. The `test` cfg of a crate is considered a different node in the dependency graph compared to its 'parent'.

Reply:
interesting, thank you! Will try to get this working then